src/types/qwcompositor.cpp

```diff
@@ -298,6 +299,11 @@ void QWSurface::unlockCached(uint32_t seq)
     wlr_surface_unlock_cached(handle(), seq);
 }
 
+void QWSurface::setFractionalScale(double scale)
+{
+    QWFractionalScaleManagerV1::notifyScale(this, scale);
```

Comment:
@zccrs

Reply:
NO, Don't add this function. You can direct use QWFractionalScaleManagerV1::notifyScale in compositor.